src/test/scala/ai/eto/rikai/sql/spark/execution/ShowModelsCommandTest.scala

```
@@ -0,0 +1,84 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.eto.rikai.sql.spark.execution
+
+import ai.eto.rikai.SparkTestSession
+import ai.eto.rikai.sql.model.{Catalog, FakeModel, Model}
+import org.scalatest.funsuite.AnyFunSuite
+
+class ShowModelsCommandTest extends AnyFunSuite with SparkTestSession {
+
+  import spark.implicits._
+
+  test("show empty") {
+    val expected =
+      Seq.empty[(String, String, String)].toDF("name", "uri", "options")
+    assertEqual(spark.sql("SHOW MODELS"), expected)
+  }
+
+  test("show models") {
+    Catalog.testing.createModel(
+      new FakeModel("model_foo", "uri://model/foo", null)
+    )
+
+    val expected =
+      Seq(("model_foo", "uri://model/foo", "{}")).toDF("name", "uri", "options")
+    val result = spark.sql("SHOW MODELS")
+    assertEqual(result, expected)
+  }
+
+  test("show models with options") {
+    spark
+      .sql(
+        "CREATE MODEL model_options OPTIONS (foo='bar',num=1.2,flag=True) USING 'fake://foo'"
+      )
+
+    val expected_options = Seq(
+      "foo" -> "bar",
+      "num" -> "1.2",
+      "flag" -> "true"
+    ).toMap
+    val expected = Seq(
+      ("model_options", "fake://foo", Model.serializeOptions(expected_options))
+    ).toDF("name", "uri", "options")
+    assertEqual(spark.sql("SHOW MODELS"), expected)
+  }
+
+  test("show multiple models") {
+    spark
+      .sql(
+        "CREATE MODEL model_foo OPTIONS (foo='bar',num=1.2,flag=True) USING 'fake://foo'"
+      )
+    assert(spark.sql("SHOW MODELS").count() == 1)
+
+    spark
+      .sql(
+        "CREATE MODEL model_bar OPTIONS (foo='bar',num=1.2,flag=True) USING 'fake://bar'"
+      )
+    assert(spark.sql("SHOW MODELS").count() == 2)
+
+    // same name
+    spark
+      .sql("CREATE MODEL model_foo USING 'fake://foo2'")
```

hmm, this supposes to raise an Exception ? or we should start to implement `CREATE OR REPLACE MODEL`

---

Yes. Let's raise an exception. And start implement `CREATE OR REPLACE MODEL`